rules/nat.go

```diff
@@ -26,13 +27,39 @@ func (r *DefaultRuleRenderer) NATOutgoingChain(natOutgoingActive bool, ipVersion
 		ipConf := r.ipSetConfig(ipVersion)
 		allIPsSetName := ipConf.NameForMainIPSet(IPSetIDNATOutgoingAllPools)
 		masqIPsSetName := ipConf.NameForMainIPSet(IPSetIDNATOutgoingMasqPools)
-		rules = []iptables.Rule{
-			{
-				Action: iptables.MasqAction{},
-				Match: iptables.Match().
-					SourceIPSet(masqIPsSetName).
-					NotDestIPSet(allIPsSetName),
-			},
+		if r.Config.NATPortRange.MaxPort > 0 {
+			toPorts := fmt.Sprintf("%d-%d", r.Config.NATPortRange.MinPort, r.Config.NATPortRange.MaxPort)
+			rules = []iptables.Rule{
+				{
+					Action: iptables.MasqAction{ToPorts: toPorts},
+					Match: iptables.Match().
+						SourceIPSet(masqIPsSetName).
+						NotDestIPSet(allIPsSetName).
+						Protocol("tcp"),
+				},
+				{
+					Action: iptables.MasqAction{ToPorts: toPorts},
+					Match: iptables.Match().
+						SourceIPSet(masqIPsSetName).
+						NotDestIPSet(allIPsSetName).
+						Protocol("udp"),
+				},
+				{
```

Comment:
Just occurred to me, is MASQUERADE a "final" action; once you've hit the tcp rule above, will the packet also hit this rule and get double NATted, or something?

Reply:
Hmm, good question. I obviously thought it was. Let me verify that...

Reply:
Of course its not. Good catch! I updated the code to exit early for tcp and udp.